### ISSUE TITLE
redundantSelf incorrectly removes self for instance variables after local variables.

### ISSUE THREAD

**Issue by CraigSiemens** [nicklockwood/SwiftFormat#1070]
If I have this file

```swift
// Foo.swift
import Foundation

extension MyStruct {
    init() {
        self.id = UUID()

        let someValue = 1234
        self.amount = someValue
    }
}
```

When I run the following command, I'd expect there to be no lint warnings.

```
swiftformat --lint foo.swift --self init-only --swiftversion 5.4
```

Instead it gets a warning only for the second property that it shouldn't use `self.`

```
Running SwiftFormat...
(lint mode - no files will be changed.)
.../foo.swift:9:1: warning: (redundantSelf) Insert/remove explicit self where applicable.
SwiftFormat completed in 0.93s.
```

Also
- If the `let someValue` is commented out then there is no warning.
- If the `let someValue` is moved to the top of init, then there is a warning for both `self.id` and `self.amount`.

**Comment by nicklockwood** [comment]
@CraigSiemens thanks for reporting. Seems to only happen with `--swiftversion` set to 5.4 or higher.

**Comment by nicklockwood** [comment]
@CraigSiemens fixed in 0.49.1
